File: Behavioral/Strategy/java/CreditCardPayment.java
package com.designpatterns.strategy;

/**
 * 信用卡支付策略
 * 具體策略實作：使用信用卡進行支付
 */
public class CreditCardPayment implements PaymentStrategy {
    private final String cardNumber;
    private final String cvv;

    public CreditCardPayment(String cardNumber, String cvv) {
        this.cardNumber = cardNumber;
        this.cvv = cvv;
    }

    @Override
    public void pay(double amount) {
        System.out.println("使用信用卡支付 $" + amount);
        System.out.println("卡號: " + cardNumber);
        System.out.println("CVV: " + cvv);
        System.out.println("支付成功！");
    }
} 

File: Behavioral/Strategy/cpp/CreditCardPayment.cpp
#include "CreditCardPayment.h"
#include <iostream>

CreditCardPayment::CreditCardPayment(const std::string& cardNumber, const std::string& cvv)
    : cardNumber(cardNumber), cvv(cvv) {
}

void CreditCardPayment::pay(double amount) {
    std::cout << "使用信用卡支付 $" << amount << std::endl;
    std::cout << "卡號: " << cardNumber << std::endl;
    std::cout << "CVV: " << cvv << std::endl;
    std::cout << "支付成功！" << std::endl;
} 